SYCL/SubGroup/helper.hpp

```diff
@@ -98,7 +109,7 @@ template <typename T2> struct utils<T2, 16> {
 
 template <typename T> void exit_if_not_equal(T val, T ref, const char *name) {
   if (std::is_floating_point<T>::value) {
-    if (std::fabs(val - ref) > 0.01) {
+    if (std::fabs(val - ref) > 0.02) {
```

Comment:
What was the reason to increase the threshold?

Reply:
> What was the reason to increase the threshold?

During data verification all elements of vectors are added and this value is compared with reference one. If we increase number of added elements twice potential cumulative error is increased twice.

Reply:
Won't it be better to switch to validating relative error instead of absolute one?

Reply:
The test does not target specific accuracy goals. It checks that the return values are not something completely different from expected.
Does it make sense to invest in tuning accuracy of the test?

Reply:
I suggest we use input data, so that results have no error at all.

Reply:
@bader, could you, please, describe a bit? I think I didn't get it.

Reply:
>During data verification all elements of vectors are added and this value is compared with reference one. 

When you do FP addition, the computed result is rounded to fit into resulting data type. The error occurs **only** if `add` result can't be exactly preserved and has to be rounded. E.g. `T == float`, `2^{20} + 2^{-10}` can't be represented "exactly", but `2^{20} + 2^{21}` can. I suggest using input values, so that rounding error will be 0, so you can always use exact match. 

BTW, using `std::fabs` here reduces accuracy for `T == double`.

Reply:
Implemented bitwise comparison from floating point type except half.
Valid range for half type is too narrow and will require to rework data for several tests. This is out of scope for current PR. But aligning input data for current test allow to revert threshold increase.